src/core/lib/surface/channel.h

```diff
@@ -174,6 +176,7 @@ class Channel : public RefCounted<Channel>,
   MemoryAllocator allocator_;
   std::string target_;
   const RefCountedPtr<grpc_channel_stack> channel_stack_;
+  const ChannelArgs channel_args_;
```

Comment:
I'd like input from @ctiller here, but it seems like it could significantly increase memory usage to store this here -- and it would affect both the client and server side, even if OT is not configured.

It doesn't seem like it would be very hard to add the layer we'd need to avoid this.  Just have `ServerCallTracerFactory` return a separate object for the connection, which we pass the channel args to and then save in the OT server filter.  Then we have *that* object return the `ServerCallTracer`.  That way, there'd be less memory increase, and it would happen only if OT is used.

Reply:
oh I think I've got a smarter method. Instead of getting it from the xds server config fetcher, I can add it to the server's channel args, and I think those are already available. Let me try it out

Reply:
The new way works without adding any additional memory requirements.

> Just have ServerCallTracerFactory return a separate object for the connection, which we pass the channel args to and then save in the OT server filter.

Note that we don't have a filter for OT on the server side. If we had one, I could have just used the channel args from that filter. Note that I still want to add a layer for a `Server` scope, but it seems like it's not needed for this PR anymore :)